gtdynamics/kinematics/Kinematics.h

```diff
@@ -87,18 +87,18 @@ class Kinematics : public Optimizer {
   /**
    * @fn Constructor.
    */
-  Kinematics(const Robot& robot,
-             const boost::shared_ptr<const KinematicsParameters>& parameters =
+  Kinematics(const boost::shared_ptr<const KinematicsParameters>& parameters =
                  boost::make_shared<const KinematicsParameters>())
-      : Optimizer(robot, parameters), p_(parameters) {}
+      : Optimizer(parameters), p_(parameters) {}
```

Comment:
Do we need `p_` in `Kinematics` since we can access it from the base class? Only downside is the constant typecasting, but that overhead shouldn't be high if we do a static cast.

Reply:
I debated about this. I will do the static cast thing, it's cleaner.